release/Makefile

```diff
@@ -0,0 +1,107 @@
+# Needed for glob expansion.
+SHELL = /bin/bash
+.SHELLFLAGS = -O extglob -c
+
+GIT_VERSION = $(shell git describe --dirty --always --tags)
```

Comment:
Since we're passing `GO_BUILD_ARGS` from the top-level Makefile, would it also make sense to pass `GIT_VERSION` in to avoid duplication?

Reply:
Yes, although I'll be adding this back once this `release` rule is updated to invoke `goreleaser`, since at that point the `release` target will be gone from the top-level Makefile.

Reply:
:+1:

> the release target will be gone from the top-level Makefile.

I'll withhold judgement about this part until I see this PR, but this piques my interest :slightly_smiling_face: